### ISSUE TITLE
Add integration with GitLab

### ISSUE THREAD

**Issue by eeriksp** [google/docsy#375]
Currently Docsy has very good integration with GitHub including but not being limited to "edit this page" and "create child page" links.

Would it be possible to implement the same for GitLab? Or at least a subset of the features should there be any restrictions on GitLab side. Something similar has already been done in the [Vuepress](https://vuepress.vuejs.org/) project.

**Comment by haslersn** [comment]
I would like this!

**Comment by ZhengHe-MD** [comment]
Yes, even if it is not fully supported now, at least it can be put into documents?

**Comment by narrenfrei** [comment]
We have also our docs hosted on GitLab and can report:

- Edit the page works fine.
- Create child page the generated URL needs to get adopted a little bit, but it should work. I have to say, that I don't use it in my project (and haven't tested to define a template for the new pages).
  - Sidenote: Also on GitHub you can also create a child page, if the current page has is a "_index.md" and has it's own folder. If it's a "normal" page without a separate folder, you create a sibling page, not a child page -i think.
- Also for creating a documentation issue for the specific page the generated URL needs to get adopted, but it should work.
- The functionality to create a project issue I've already adopted for GitLab and it works fine.
